Post Terms: Add wp_kses_post to prefix/suffix for consistency

andrewserong opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Following on from #40776 and #40559, this adds `wp_kses_post` to the Post Terms prefix and suffix fields.

Also:

* Add `allowedFormats` to reduce the allowed formats for consistency with other blocks like Navigation Link (we probably only really need bold, italic, image, strikethrough rather than keyboard / super/subscript options) – I also added in link and text color, as they're probably still useful here.
* Add `__unstableOnSplitAtEnd` to the suffix field so that if someone presses enter at the end of the suffix, it adds a Paragraph block instead of a newline character to the suffix field.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

This probably isn't _really_ necessary, but for consistency with the other blocks that have RichText fields and server-render those fields, I thought we could add it in. (Happy to close this out if folks think it doesn't need it).

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

Wrap the prefix and suffix in `wp_kses_post`

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

On a post with tags attached, add a Post Tags block, and enter text in the prefix or suffix fields. Using the RichText controls, set some of the text to bold or italic and ensure it still renders correctly on the frontend.

## Screenshots or screencast <!-- if applicable -->

![image](https://user-images.githubusercontent.com/14988353/166622187-17728918-b4d7-463b-86b1-87567b2b9c7e.png)

andrewserong committed:
Post Terms: Add wp_kses_post for consistency

ntsekouras reviewed (approved):
Thanks for the PR @andrewserong!

This doesn't hurt of course, but since we are not disabling the formats in the RichText instances used in `edit` function, the formats are preserved and other html is treated as `string`. I'm approving and leave this to you 😄

andrewserong committed:
Add allowedFormats list to Post Terms, and __unstableOnSplitAtEnd to create a paragraph block when hitting enter at the end of the Suffix field

andrewserong committed:
Move allowedFormats to a constant, add in text color and link

andrewserong commented:
Thanks for the reviews, and the feedback @ntsekouras! I've added in `allowedFormats` in the `Edit` component, and used a similar list as in Navigation Link (but also included link and text color). Judging by #40559, it looks like the intention of the prefix / suffix fields is for fairly simple inline text, so a fairly limited option here makes sense to me. @dsas if we want additional options here, we can always add some back in in a follow-up!

While I was editing that file, I also added in `__unstableOnSplitAtEnd` to the suffix field so that if you're in the suffix field and hit Enter, it creates a paragraph block below, like on many other blocks, to avoid accidentally getting stuck adding newline characters in the suffix field (users can still add a newline character if need be by holding the `shift` key).

I'll merge this in once tests pass.
